cmake/CableCompilerSettings.cmake

```diff
@@ -0,0 +1,168 @@
+
+            # Stack protection.
+            check_cxx_compiler_flag(-fstack-protector fstack-protector)
+            if(fstack-protector)
+                # The compiler supports stack protection options.
+                if(cable_NO_STACK_PROTECTION)
+                    # Stack protection explicitly disabled.
+                    # Add "no" flag, because in some configuration the compiler has it enabled by default.
+                    add_compile_options(-fno-stack-protector)
+                else()
+                    # Try enabling the "strong" variant.
+                    cable_add_cxx_compiler_flag_if_supported(-fstack-protector-strong have_stack_protector_strong_support)
+                    if(NOT have_stack_protector_strong_support)
+                        # Fallback to standard variant of "strong" not available.
+                        add_compile_options(-fstack-protector)
+                    endif()
+                endif()
+            endif()
+
+            cable_add_cxx_compiler_flag_if_supported(-Wimplicit-fallthrough)
+
+        elseif(MSVC)
+
+            # Get rid of default warning level.
+            string(REPLACE " /W3" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+            string(REPLACE " /W3" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+
+            # Enable basing warnings set and treat them as errors.
+            add_compile_options(/W4 /WX)
+
+            # Allow unknown pragmas, we don't want to wrap them with #ifdefs.
+            add_compile_options(/wd4068)
+
+        endif()
+
+        # Option for arch=native.
+        option(NATIVE "Build for native CPU" OFF)
+        if(NATIVE)
+            if(MSVC)
+                add_compile_options(-arch:AVX)
+            else()
+                add_compile_options(-mtune=native -march=native)
+            endif()
+        elseif(NOT MSVC)
+            # Tune for currently most common CPUs.
+            cable_add_cxx_compiler_flag_if_supported(-mtune=generic)
+        endif()
+
+        # Sanitizers support.
+        set(SANITIZE OFF CACHE STRING "Build with the specified sanitizer")
+        if(SANITIZE)
+            # Set the linker flags first, they are required to properly test the compiler flag.
+            set(CMAKE_SHARED_LINKER_FLAGS "-fsanitize=${SANITIZE} ${CMAKE_SHARED_LINKER_FLAGS}")
+            set(CMAKE_EXE_LINKER_FLAGS "-fsanitize=${SANITIZE} ${CMAKE_EXE_LINKER_FLAGS}")
+
+            set(test_name have_fsanitize_${SANITIZE})
+            check_cxx_compiler_flag(-fsanitize=${SANITIZE} ${test_name})
+            if(NOT ${test_name})
+                message(FATAL_ERROR "Unsupported sanitizer: ${SANITIZE}")
+            endif()
+            add_compile_options(-fno-omit-frame-pointer -fsanitize=${SANITIZE})
+
+            set(backlist_file ${PROJECT_SOURCE_DIR}/sanitizer-blacklist.txt)
+            if(EXISTS ${backlist_file})
+                check_cxx_compiler_flag(-fsanitize-blacklist=${backlist_file} have_fsanitize-blacklist)
```

Comment:
seems these lines here could have been replaced with `cable_add_cxx_compiler_flag_if_supported`

Reply:
I will try that. We can also drop this option, because it comes from old Aleth config and is not used in any project currently.